Packs/GenericSQL/Integrations/GenericSQL/GenericSQL.py

```diff
@@ -282,11 +291,21 @@ def sql_query_execute(client: Client, args: dict, *_) -> Tuple[str, Dict[str, An
         bind_variables = generate_bind_vars(bind_variables_names, bind_variables_values)
 
         result, headers = client.sql_query_execute_request(sql_query, bind_variables)
-        # converting an sqlalchemy object to a table
-        converted_table = [dict(row) for row in result]
-        # converting b'' and datetime objects to readable ones
-        table = [{str(key): str(value) for key, value in dictionary.items()} for dictionary in converted_table]
+
+        if client.dialect == "Teradata":
+            table = []
+            for row in result:
+                convert_tup_to_dic = {header: col for header, col in zip(headers, row)}
+                table.append(convert_tup_to_dic)
+        else:
+            # converting a sqlalchemy object to a table
+            converted_table = [dict(row) for row in result]
+            # converting b'' and datetime objects to readable ones
+            table = [{str(key): str(value) for key, value in dictionary.items()} for dictionary in converted_table]
```

Comment:
the `dictionary` is actually a table row, right? let's call it that, to make debugging easier

Reply:
Done